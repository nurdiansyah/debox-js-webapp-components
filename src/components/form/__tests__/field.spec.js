import React from 'react'

import {Form, Field, getFormBlock} from '../index'

describe('form/field', () => {
  it('should when not label', () => {
    const formBlock = getFormBlock({})
    const wrapper = render(
      <Form block={formBlock}>
        <Field block={formBlock}>
          <div>test</div>
        </Field>
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should with label', () => {
    const formBlock = getFormBlock({})
    const wrapper = render(
      <Form block={formBlock}>
        <Field block={formBlock} label="label test">
          <div>test</div>
        </Field>
      </Form>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
