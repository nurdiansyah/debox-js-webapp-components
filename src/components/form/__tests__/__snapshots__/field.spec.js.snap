// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form/field should when not label 1`] = `
<form
  class="form"
>
  <div
    class="form__field  "
  >
    <div>
      test
    </div>
  </div>
</form>
`;

exports[`form/field should with label 1`] = `
<form
  class="form"
>
  <div
    class="form__field  "
  >
    <label
      class="form__label"
    >
      label test
    </label>
    <div>
      test
    </div>
  </div>
</form>
`;
